### DISCUSSION TITLE
Coal emission factors (for ENTSO-E data)

### DISCUSSION THREAD

**Discussion by mirkoschaefer** [electricitymaps/electricitymaps-contrib#4934] [General]
The current emission factors might overestimate emissions from coal for instance for Germany. In your [method ](https://colab.research.google.com/drive/1buA-yHQCF711F53IX7_GAxYMJ9p4qJoT?usp=sharing) you aggregate the following technologies to coal:
Fossil Brown coal/Lignite, Fossil Coal-derived gas, Fossil Hard coal, Fossil Oil shale, Fossil Peat
With this aggregation there are two issues:
i) How to categorize Fossil Coal-derived gas, Fossil Oil shale and Fossil Peat
ii) How to discriminate between lignite and hard coal (significant generation for Germany, for instance)

i) Fossil Coal-derived gas, Fossil Oil shale and Fossil Peat could be classified as other or unknown. I am not sure about this, but this should be clarified since it probably has an impact on the emission factors. The per unit generation data for Germany, for instance, includes blocks with power generation from blast furnace gas, which have a very high emission factor. Since your method includes data from a limited number of power plants only, these plants could have a disproportionally high influence on the emission factor.

ii) You aggregate lignite and hard coal, although each of these types has high generation volumes, significant different emission factors, and separate generation data is available from ENTSO-E. Ideally, you should use and publish emission factors for lignite and hard coal separately, and separate the generation data in your pipeline. If you want to use one joint emission factor for coal and aggregate hard coal and lignite in your generation data pipeline, you should compare the generation capacity/generation for lignite and hard coal in the per unit generation data used in your [method ](https://colab.research.google.com/drive/1buA-yHQCF711F53IX7_GAxYMJ9p4qJoT?usp=sharing#scrollTo=L_0oWsxygy7l) with the overall generation capacity/generation for these technologies in the different countries. Due to the limitation to power plants contained in the per unit generation data from ENTSO-E and the automatic matching procedure, there might be a considerable difference. In this case you could add corresponding scaling factors in your method. Finally, it would be helpful to compare the hourly shares of lignite/hard coal in the joint coal generation data. This would help to estimate the impact of aggregating these technologies.

**Comment by VIKTORVAV99** [comment]
There has been some discussions around this already here: https://github.com/electricitymaps/electricitymaps-contrib/discussions/4219#discussioncomment-3110126
But the team has and is very busy at the moment so we have not explored it any further at the moment as far as I know.

The suggested approach in there would allow for highly granular and hourly emission factors while keeping our current categories.

I hope this makes sense and if you with the approach I linked do feel free to raise your vote for it there as well, it will help us prioritize what to work on next.

**Reply by mirkoschaefer** [reply]
Something like that would make sense if you have power generation input data and emission factors on a technologically more granular level compared to what you want to display. I also understand that this is a more general feature and thus need more time to implement. Just to visualize the problem see the figure below. It shows the combined generation from lignite and hard coal for each hour in Germany in 2021 and 2022, sorted by the share of hard coal generation (red line shows the share of hard coal, shares are not clearly visible for the hard coal / lignite values since the resolution is not high enough to show the fluctuating hourly values) - thanks to a colleague for this plot. Using a combined factor for hard coal and lignite assumes that the read line is horizontal. 

In any case you could check how the shares of hard coal are in the generation data you use for the emission factor calculation. I didn't run your notebook, but looking through it you have 35 EU ETS entries for coal in Germany in your matching. I assume your positive outliers are the blast furnace gas plants, your negative outlier could be the Boxberg plant, which has two EU ETS entries (an issue I just noticed in our data set). The remaining ones could show bias towards lignite power plants, so the hypothetical horizontal red line not only ignores hourly fluctuations in the hard coal / lignite shares, but probably also is too high.

PS: For checking your emission factors it would also be helpful if you provide the final power plant list (maybe with generation and emission data) from your calculation, like you already do it with the manually matched plants in "plants_mapped_manually.csv".

![hardcoal_lignite](https://user-images.githubusercontent.com/57491780/211298612-9eeff6d3-f9fc-45cb-9509-c9450d0041d1.png)

**Reply by VIKTORVAV99** [reply]
@pierresegonne is this something you could shed some light on?

**Comment by pierresegonne** [comment]
Hey @mirkoschaefer, sorry for taking so long to answer!

> How to categorize Fossil Coal-derived gas, Fossil Oil shale and Fossil Peat?

I think we could adopt a per type approach her. I believe that the IEA sets everything that is derived from coal to coal, so we could attribute fossil coal-derived gas as coal.

For oil shale and peat that's a bit more tricky. The best would indeed be to include them into unknown but that opens the door to a lot of complexity. I say this because it then means we also need to compute zone specific emission factors for unknown, that accounts for all unknown (and thus potentially stuff that is not reported in the EU-ETS scheme)

Else we can just exclude them from the computation, do you think they can have a significant impact on the final emission factors? We're weighting everything by the total production, so my intuition is that it does not. 

> Hard coal vs lignite:

Many thanks for the plot, not being able to separate the types of coal does creates some issues at an hourly level (not yearly because aggregated we should end up with the same). I think this issue, and the amount of people not understanding that the computed emission factor accounts for the average mixup is a good argument for splitting up coal. The only issue is that it will require substantial work and we can't roll it out instantly. 

I guess a first step could be to refine the German data using some of your work on matching all of the German plants, which we could do in a more reasonable amount of time. 

Thanks for the final suggestion, we'll do that once we update the computation notebook :)

**Reply by mirkoschaefer** [reply]
Hey @pierresegonne , thanks for your answer.

It is helpful to consider the type of data you are using to estimate the impact of these methodological decisions. For your hourly signal, you are mainly using ENTSO-E **per_type** generation data. The per_type generation data only represents a certain share of the actual generation of a country, due to the reporting schemes of the different TSOs (only a certain share of generators report their data, there can also be misclassifications depending on the country, etc.). This is a fundamental problem which would need fixing on a per country basis ([Agorameter ](https://www.agora-energiewende.de/projekte/agorameter/)does this fixing in an elaborate way for Germany, for instance). If you now calculate, for instance, the average emission intensity for Germany for 2020 based on your data (which makes use of hourly per_type data) and compare it with the reported emission intensity of Germany as published by the [EEA](https://www.eea.europa.eu/data-and-maps/daviz/co2-emission-intensity-12#tab-googlechartid_chart_11), for instance, you will get different results. Because the published annual emission intensity makes use of reported statistical data (from Eurostat), whereas you use hourly generation data from ENTSO-E. We discussed this in our [paper ](https://www.sciencedirect.com/science/article/pii/S2772783122000176)here, for some issues comparing these data sets see also the discussion [here](https://forum.openmod.org/t/matching-entso-e-production-type-and-eurostat-siec-data-categories/3637). 

Now, for the problem discussed here, you need to discriminate hourly **per_type** generation data from ENTSO-E, and **per_unit** generation data from ENTSO-E. In my understanding, you use per_unit data for the calculation of emission factors, whereas you use per_type data to calculate the actual hourly signal. Generally, the per_type data is not identical to the aggregated per_unit data. So if you sum up all per_unit generation data for a certain type (hard coal, for instance) and compare it with the corresponding per_type data, you get different numbers. Generally you get less, because the coverage of the per_unit data is lower than the coverage of the per_type data (which again is lower than the coverage of the overall generation represented by Eurostat, as discussed above). Additionally there might be issues with classification in the ENTSO-E data, I recall that for a certain type and country the sum over all units is larger than the corresponding per_type data.

In your calculation, you calculate emission factors based on per_unit generation data. So if you would keep the original classification (each type apart), you still would need to check if your per_unit data (for emission factor calculation) is representative for the corresponding per_type generation (which you later use for your actual signal). For Germany and fossil gas, for instance, the per_unit data only covers a certain share of the per_type data (order of 60%, I'd guess), because a lot of smaller plants don't report per_unit data, but are representend in the per_type data.

Since you are grouping different types in your classification, you additionally have to check if the resulting mix in your per_unit data is the same as in your per_type data. This affects directly your statement above that "aggregated we should end up with the same" - this statement only holds if the aggregated per_unit data is identical to the corresponding per_type data (and would stay the same over time). I don't think this is the case. For instance, your "other" generation is very likely to have a small share in your per_type data, but it might have a higher share in your per_unit data, and therefore affects the calculation of emission factors and therefore your signal. Now for hard-coal and lignite, I'd guess that the share of lignite in your per_unit data is higher than in your per_type data, which would lead to a bias to higher emissions for coal (only a guess, I would need to check myself).

My suggestion would be to carefully compare your aggregated per_unit data and per_type data, and then maybe, as a first fix, calculate emission factors for each type separately and then weight them with historical per_type generation.

**Reply by pierresegonne** [reply]
Hey @mirkoschaefer,

Indeed, all of these concerns are valid.

It's something we wanted to discuss last time we were supposed to discuss, but unfortunately our schedule forced us to postpone. I think we did not reschedule a precise time, so let's do that over email.

I think we could definitely draw inspiration from what you've been doing in Germany or what Agorameter has done. I hope that we can find a way to make this scalable. 

We'll keep you updated with further developments, and once we have started shaping these data corrections.

Best

**Reply by mirkoschaefer** [reply]
Hi @pierresegonne ,
not a lot to add to the points above, just wanted to share a small check I finally got around to do. In the table, you see for different technology types (ENTSO-E) the number of units in the per_unit_generation data set, their joint capacity, and their aggregated generation in 2022 (Germany only). This is the input for the per type emission factor calculation. In the last column you see the aggregated generation from the per_type_generation data set, which is the input for the overall emission factor calculations in your pipeline.
As you can see, for lignite and nuclear both numbers coincide, because these are large blocks for which per_unit_generation data is available. For hard coal, only 85% of the per_type_generation is included in the per_unit_generation data, because of smaller blocks which have to report their generation (that is why it is in the per_type), but for which no per_unit_generation data is available, for fossil gas only 57%. Then you have classification issues - for instance, fossil coal-derived gas publishes per_unit_generation data, but for the per_type_generation data it seems to be included in another category.
Also, this might change over time. For instance, in Germany in 2020 the hard-coal per_type and per_unit generation did almost coincide, but not because the per_unit generation data had more coverage, but because the per_type generation data had less coverage (as you can see if you compare the ENTSO-E data with reported yearly statistical data).
So -  it's complicated ;-) 

![per_unit_per_type](https://user-images.githubusercontent.com/57491780/229067742-f8c8dfe4-710e-4ccd-970c-b9012dc6aea3.png)

**Reply by pierresegonne** [reply]
Great!

Thanks for sharing, we were already convinced after our last discussion of the need to re-run the calculations with a correction for how representative each precise type (lignite, hard coal) is for the aggregated types (coal).

We should roll out the update in the next two months or so
